### PR TITLE
Ignore phantom contact points (negative penetration depth) from Bullet

### PR DESCRIPTION
**Pull request by mxgrey**
This PR is for investigating the issue reported in #1184.

***

**Before merging a pull request**

- [x] Set version target by selecting a milestone on the right side
- [x] Summarize this change in `CHANGELOG.md`
- [x] Add unit test(s) for this change

### CODE REVIEW

File: dart/constraint/ConstraintSolver.cpp [hunk 1]
```diff
@@ -441,7 +441,11 @@ DART_SUPPRESS_DEPRECATED_END
       mSoftContactConstraints.push_back(
             std::make_shared<SoftContactConstraint>(contact, mTimeStep));
     }
-    else
+    // If penetration depth is negative, then the collision isn't really
+    // happening and the contact point should be ignored.
+    // TODO(MXG): Investigate ways to leverage the proximity information of a
+    //            negative penetration to improve collision handling.
+    else if(contact.penetrationDepth >= 0.0)
```
Comment (jslee02) [inline dart/constraint/ConstraintSolver.cpp]: I think we would like to ignore negative penetration contacts even for soft contact liek:
```cpp
if(contact.penetrationDepth < 0.0)
  continue;

if (isSoftContact(contact))
{
  ...
}
else
{
  ...
}
```

Comment (jslee02) [reply dart/constraint/ConstraintSolver.cpp]: @mxgrey Any thoughts on this? :smile:

Comment (mxgrey) [reply dart/constraint/ConstraintSolver.cpp]: Sorry for the delay, I got sidetracked. Makes sense to me! I've updated the branch with this change.

Comment (jslee02) [reply dart/constraint/ConstraintSolver.cpp]: Thanks! Let's merge once CI tests are done.

File: dart/constraint/ConstraintSolver.cpp [hunk 2]
```diff
@@ -436,16 +436,19 @@ DART_SUPPRESS_DEPRECATED_BEGIN
     shapeFrame2->asShapeNode()->getBodyNodePtr()->setColliding(true);
 DART_SUPPRESS_DEPRECATED_END
 
+    // If penetration depth is negative, then the collision isn't really
+    // happening and the contact point should be ignored.
+    // TODO(MXG): Investigate ways to leverage the proximity information of a
+    //            negative penetration to improve collision handling.
+    if (contact.penetrationDepth >= 0.0)
```
Comment (jslee02) [inline dart/constraint/ConstraintSolver.cpp]: Oh, wait. Actually, this should be:
```cpp
if (contact.penetrationDepth < 0.0)
```
to match the original logic.

Comment (mxgrey) [reply dart/constraint/ConstraintSolver.cpp]: Whoops! Thank goodness for unit tests :sweat_smile:

**Pull request merged**